### PR TITLE
Feature/order manager class

### PR DESCRIPTION
**Pull request by Ziv-Laitman**
# Added Order Management System with Persistence and Tests

## Overview
This PR introduces the `OrderManager` class, responsible for handling order storage and retrieval. It ensures reliable persistence of order data in a JSON file while providing structured query methods. Additionally, `test_order_manager.py` includes unit tests to validate order management operations.

### **OrderManager (`order_manager.py`)**
- **Persistent Order Storage**:
  - Saves orders in a JSON file (`orders.json`).
  - Ensures structured serialization of order data, including items, pricing, and payment details.
- **Order Retrieval**:
  - Fetches orders by `order_id`.
  - Retrieves all orders associated with a given `user_id`.
- **Robust Error Handling**:
  - Ensures JSON file integrity and validation.
  - Raises meaningful exceptions for incorrect file formats or missing attributes.

### **Unit Tests (`test_order_manager.py`)**
- **Tested Core Functionalities**:
  - Successfully saves and retrieves orders.
  - Verifies orders are stored with correct attributes.
  - Ensures retrieval functions return expected results.
- **Edge Cases & Error Handling**:
  - Tests for missing or malformed JSON data.
  - Validates order retrieval with nonexistent `order_id` or `user_id`.

## Why This Change?
- Provides a **centralized order management system** with persistent storage.
- Ensures **reliable data retrieval and serialization** for order tracking.
- Improves **test coverage** to prevent data inconsistencies.

## Next Steps
- Integrate order management with the API to allow external interactions.
- Expand tests to handle concurrency and file corruption scenarios.

---

### **Reviewers**
- Verify that order serialization and retrieval function correctly.
- Ensure that the file handling operations align with project requirements.
- Confirm test coverage includes all key functionalities.

Looking forward to your feedback!

### CODE REVIEW

File: app/models/order_manager.py
Comment (TamarAdn) [inline app/models/order_manager.py:1]: Great OrderManager implementation! The separation of concerns with JsonFileManager is super clean.

File: app/models/order_manager.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+from typing import Any, Dict, List, Optional, cast
+
+from app.config import ORDERS_FILE
+from app.utils import JsonFileManager
+
+
+class OrderManager:
+    """
```
Comment (TamarAdn) [inline app/models/order_manager.py:8]: Great job with the docstrings and type annotations!

File: app/models/order_manager.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        Returns:
+            The order dictionary if found, None otherwise.
+
+        Raises:
+            TypeError: If the file contents cannot be processed.
+        """
+        orders = cast(List[Dict[str, Any]], JsonFileManager.read_json(self._file_path))
+
+        if not isinstance(orders, list):
```
Comment (TamarAdn) [inline app/models/order_manager.py:77]: Proper exception raising with specific error messages

File: tests/models/test_order_manager.py
Comment (TamarAdn) [inline tests/models/test_order_manager.py:1]: - Very clear to the reader 
- Proper use of pytest.fixture and @pytest.mark.parametrize 

I think the tests are very good and cover what we need

File: app/models/order_manager.py
Comment (TamarAdn) [inline app/models/order_manager.py:1]: The query methods provide a good foundation for our order retrieval needs. 
The separation of Order and OrderManager classes demonstrates excellent adherence to the Single Responsibility Principle.
